Documentation/proposals/OneDotNetEmbeddedResources.md

```diff
@@ -0,0 +1,247 @@
+* `@(AndroidNativeLibrary)` - `.so` file to include in an application
+  project.
+
+Let's simplify this, we could support all of the above with a new
+`@(AndroidLibrary)` item group:
+
+```xml
+<!-- Include and bind -->
+<AndroidLibrary Include="foo.aar" Bind="true" />
+<!-- Just include, do not bind -->
+<AndroidLibrary Include="bar.aar" />
+<AndroidLibrary Include="baz.jar" />
+<!--
+  Bind but do not include in NuGet package.
+  %(Pack) is built into NuGet MSBuild targets.
+-->
+<AndroidLibrary Include="bar.aar" Bind="true" Pack="false" />
+<!-- Native libraries need ABI directory -->
+<AndroidLibrary Include="armeabi-v7a\libfoo.so" />
+<AndroidLibrary Include="x86\libfoo.so" />
+```
+
+The new `@(AndroidLibrary)` item group will simply translate to the
+old ones for backwards compatibility. The extension of the file can be
+used to determine what kind of library each item is. `%(Bind)` will be
+`false` by default, and `%(Pack)` will be `true` by default. `%(Pack)`
+will not do anything in application projects.
+
+The deprecated item groups will no longer embed, but pack into
+`.nupkg` files instead:
+
+* `@(EmbeddedJar)`
+* `@(EmbeddedReferenceJar)`
+
+`@(LibraryProjectZip)` will not be supported at all.
+
+## Implementation
+
+Since there is a lot of work here, I've split up the what needs to be
+done into different phases.
+
+### "Must Have" for .NET 6 Preview 1
+
+Things we need for the first public release:
+
+* Item group support for `@(AndroidLibrary)` and `%(Bind)=true`.
+* Support for packing `.aar` files in `$(OutputPath)` for
+  Xamarin.Android class libraries.
+* Support for consuming `.nupkg` files that contain `.aar` files.
+
+These should enable us to compile AndroidX for `net6.0-android`. Then
+potentially any other library that depends on AndroidX.
+
+### Next Phase
+
+Things that could happen by .NET 6 RTM:
+
+* Support for referencing Android Studio projects in order to build
+  them and consume the resulting `.aar` file.
+
+* Support for including Maven dependencies. This might be inclusion of
+  a `build.gradle` file that would enable us to run `gradle` and
+  download dependencies from Maven repositories.
+
+* Consider not extracting everything in `.aar` files. We can probably
+  skip the `assets` and `res` directories and let `aapt2` consume
+  `.aar` files directly.
+
+* `<GenerateJavaStubs/>` performance
+  * `.aar` files contain `AndroidManifest.xml` files. We could add
+    support for developers to add their own manifest to Xamarin.Android
+    class libraries.
+  * We could also generate the `AndroidManifest.xml` so attributes
+    like `[assembly:UsesPermission]` are done ahead of time.
+  * We could generate Java stubs for every `Java.Lang.Object`
+  * If all of the above was completed, `<GenerateJavaStubs/>` could
+    skip over the assembly completely.
+
+* Build performance could be further improved by included various
+  other build-related cache files in the `.aar` file.
```

Comment:
@dellis1972 I tried to put several ideas on how we could use the `.aar` file to improve build performance in the future.